File: src/main/java/model/rules/blokedgame/SmallestHandOnePoint.java
package model.rules.blokedgame;

import model.players.Player;
import model.RoundResult;
import java.util.ArrayList;
import java.util.List;

public class SmallestHandOnePoint implements BlockedGame {

    @Override
    public RoundResult execute(List<Player> players) {
        List<Integer> sums = new ArrayList<>();
        players.stream().map(p -> p.getTiles()).forEach(tiles -> sums.add(tiles.stream().mapToInt(t -> t.sum()).sum()));
        int min = sums.stream().min(Integer::compare).get();
        long count = sums.stream().filter(p -> p == min).count();
        boolean sameCouple = (sums.get(0) == min && sums.get(2) == min) || (sums.get(1) == min && sums.get(3) == min);
        if (count > 2 || (count == 2 && !sameCouple)) {
            return new RoundResult(RoundResult.DRAW, 0);
        }
        return new RoundResult(sums.indexOf(min), 1);
    }
    
}

File: docs/scripts/model/rules/blockedgame/SmallestHandOnePoint.js
import Result from "../../Result.js";
import {ResultType} from "../../ResultType.js";

export default class SmallestHandOnePoint {

    execute(players) {
        let sums = [];
        players.map(p => p.getTiles()).forEach(tiles => sums.push(tiles.map(t => t.sum()).reduce((a, b) => a + b, 0)));
        let min = Math.min(...sums);
        let count = sums.filter(p => p === min).length;
        let sameCouple = (sums.get(0) === min && sums.get(2) === min) || (sums.get(1) === min && sums.get(3) === min);
        if (count > 1 && !sameCouple) {
            return new Result(ResultType.DRAW);
        }
        return new Result(sums.indexOf(min), 1);
    }
}
